src/Campaigns/Controllers/CampaignRequestController.php

```diff
@@ -0,0 +1,159 @@
+
+        $query
+            ->limit($perPage)
+            ->offset(($page - 1) * $perPage);
+
+        $campaigns = $query->getAll() ?? [];
+        $totalCampaigns = empty($campaigns) ? 0 : Campaign::query()->count();
+        $totalPages = (int)ceil($totalCampaigns / $perPage);
+
+        $response = rest_ensure_response($campaigns);
+        $response->header('X-WP-Total', $totalCampaigns);
+        $response->header('X-WP-TotalPages', $totalPages);
+
+        $base = add_query_arg(
+            map_deep($request->get_query_params(), function ($value) {
+                if (is_bool($value)) {
+                    $value = $value ? 'true' : 'false';
+                }
+
+                return urlencode($value);
+            }),
+            rest_url(CampaignRoute::CAMPAIGNS)
+        );
+
+        if ($page > 1) {
+            $prevPage = $page - 1;
+
+            if ($prevPage > $totalPages) {
+                $prevPage = $totalPages;
+            }
+
+            $response->link_header('prev', add_query_arg('page', $prevPage, $base));
+        }
+
+        if ($totalPages > $page) {
+            $nextPage = $page + 1;
+            $response->link_header('next', add_query_arg('page', $nextPage, $base));
+        }
+
+        return $response;
+    }
+
+    /**
+     * @unreleased
+     *
+     * @return WP_Error | WP_REST_Response
+     *
+     * @throws Exception
+     */
+    public function updateCampaign(WP_REST_Request $request)
+    {
+        $campaign = Campaign::find($request->get_param('id'));
+
+        if ( ! $campaign) {
+            return new WP_Error('campaign_not_found', __('Campaign not found', 'give'), ['status' => 404]);
+        }
+
+        $statusMap = [
+            'draft' => CampaignStatus::DRAFT(),
+            'active' => CampaignStatus::ACTIVE(),
+        ];
+
+        foreach ($request->get_params() as $key => $value) {
+            switch ($key) {
+                case 'id':
+                    break;
+                case 'status':
+                    $status = array_key_exists($value, $statusMap)
+                        ? $statusMap[$value]
+                        : CampaignStatus::DRAFT();
+
+                    $campaign->setAttribute('status', $status);
+
+                    break;
+                default:
+                    if ($campaign->hasProperty($key)) {
+                        $campaign->setAttribute($key, $value);
+                    }
+            }
+        }
+
+        if ($campaign->isDirty()) {
+            $campaign->save();
+        }
+
+        return new WP_REST_Response($campaign->toArray());
+    }
+
+
+    /**
+     * @unreleased
+     *
+     * @throws Exception
+     */
+    public function createCampaign(WP_REST_Request $request): WP_REST_Response
+    {
+        $campaign = Campaign::create([
+            'type' => CampaignType::CORE(),
+            'title' => $request->get_param('title'),
+            'shortDescription' => $request->get_param('shortDescription'),
```

Comment:
As this field is not required on the UI, I think we can replace it by this:  `$request->get_param('shortDescription') ?? ''`